storage/sql/sql.go

```diff
@@ -59,19 +59,34 @@ var (
 		// NOTE(ericchiang): For some reason using `SET SESSION CHARACTERISTICS AS TRANSACTION` at a
 		// session level didn't work for some edge cases. Might be something worth exploring.
 		executeTx: func(db *sql.DB, fn func(sqlTx *sql.Tx) error) error {
-			tx, err := db.Begin()
-			if err != nil {
-				return err
-			}
-			defer tx.Rollback()
+			for {
+				tx, err := db.Begin()
```

Comment:
I wonder if it would be worth switching to `BeginTx(context.TODO(), sql.LevelSerializable)`, which would, I think, mean we can ditch the explicit `SET TRANSACTION ISOLATION LEVEL SERIALIZABLE`.

Reply:
I suppose if we did that we could also go all-in on contexts and do something like
```go
ctx, cancel := context.WithCancel(context.TODO())
defer cancel() // will rollback transaction
tx, err := db.BeginTx(ctx, nil /* or some proper level? */)
if err != nil {
  return err
}
if err := fn(tx); err != nil {
  return err
}
err = tx.Commit()
// etc
```

Reply:
Neat, didn't know that was a thing! I'll try it out.